org.eclipse.xtext.ui/src/org/eclipse/xtext/ui/editor/XtextSourceViewer.java

```diff
@@ -131,13 +149,25 @@ public IContentAssistant getContentAssistant() {
 	}
 
 	/**
+	 * Supported adapter types are {@link IReconciler} and {@link IXtextDocument}.
+	 * 
 	 * @since 2.3
 	 */
 	@Override
 	public <T> T getAdapter(Class<T> adapter) {
-		if (IReconciler.class.isAssignableFrom(adapter)) {
+		if (IReconciler.class.isAssignableFrom(adapter) && adapter.isInstance(fReconciler)) {
```

Comment:
isnt the first assignable from covered by the second isInstance?

Reply:
The second would return true for `getAdapter(Object.class)` which would be a surprise.